test/Set.test.mo

```diff
@@ -1907,4 +1909,83 @@ run(
 
     ]
   )
+);
+
+Test.suite(
+  "valuesFrom",
+  func() {
+    Test.test(
+      "Simple",
+      func() {
+        let set = Set.empty<Nat>();
+        Set.add(set, Nat.compare, 1);
+        Set.add(set, Nat.compare, 2);
+        Set.add(set, Nat.compare, 4);
+        func check(from : Nat, expected : [Nat]) {
+          let actual = Iter.toArray(Set.valuesFrom(set, Nat.compare, from));
+          Test.expect.array(actual, Nat.toText, Nat.equal).equal(expected)
+        };
+        check(0, [1, 2, 4]);
+        check(1, [1, 2, 4]);
+        check(2, [2, 4]);
+        check(3, [4]);
+        check(4, [4])
+      }
+    );
+    Test.test(
+      "Extensive 2D test",
+      func() {
+        let set = Set.empty<Nat>();
+        let n = 100;
+        for (i in Nat.rangeBy(1, n, 2)) {
+          Set.add(set, Nat.compare, i);
+          for (j in Nat.range(0, i + 2)) {
+            let actual = Iter.toArray(Set.valuesFrom(set, Nat.compare, j));
+            let expected = Iter.toArray(Iter.dropWhile<(Nat)>(Set.values(set), func(k) = k < j));
+            Test.expect.array(actual, Nat.toText, Nat.equal).equal(expected)
+          }
+        }
+      }
+    )
+  }
+);
+
+Test.suite(
+  "reverseValuesFrom",
+  func() {
+    Test.test(
+      "Simple",
+      func() {
+        let set = Set.empty<Nat>();
+        Set.add(set, Nat.compare, 1);
+        Set.add(set, Nat.compare, 2);
+        Set.add(set, Nat.compare, 4);
+        func check(from : Nat, expected : [Nat]) {
+          let actual = Iter.toArray(Set.reverseValuesFrom(set, Nat.compare, from));
+          Test.expect.array(actual, Nat.toText, Nat.equal).equal(expected)
+        };
+        check(0, []);
+        check(1, [1]);
+        check(2, [2, 1]);
+        check(3, [2, 1]);
+        check(4, [4, 2, 1]);
+        check(5, [4, 2, 1])
+      }
+    );
+    Test.test(
+      "Extensive 2D test",
+      func() {
+        let set = Set.empty<Nat>();
+        let n = 100;
```

Comment:
Same here, try with n = 1000

Reply:
I did test manually with higher `n` in both cases. Also with smaller `btreeOrder` like `4` to cover more tree-shapes. Unfortunately the order is not customizable, otherwise I'd leave them in tests.
And yes I left `n=100` so that it's faster but still `>32`